lib/beaker-pe/install/pe_utils.rb

```diff
@@ -485,7 +485,7 @@ def simple_monolithic_install(master, agents, opts={})
           end
 
           step "Run puppet to setup mcollective and pxp-agent" do
-            on(master, puppet_agent('-t'), :acceptable_exit_codes => [0,2])
+            on(master, puppet_agent('-t'), :acceptable_exit_codes => [2])
```

Comment:
Damn, you're right.

The solution to that would be to re-order so we ensure the install completes before we start mucking around with agents.  Which I think is also best.